admin/helpers/school-home-pin-generation-eligibility-presenter.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const moment = require('moment')
+
+const checkWindowV2Service = require('../services/check-window-v2.service')
+const config = require('../config')
+const monitor = require('../helpers/monitor')
+
+const schoolHomePinGenerationEligibilityPresenter = {}
+
+/**
+ * Fetch data for familiarisation and live pin generation eligibility
+ * @returns {Object} Eligibility data including flags and relevant datetimes
+ */
+schoolHomePinGenerationEligibilityPresenter.getPresentationData = async () => {
+  const checkWindow = await checkWindowV2Service.getActiveCheckWindow()
+  const overridePinGenerationEligibility = config.OverridePinExpiry
+  const currentDate = moment.utc()
+  const pinGenerationEligibilityData = {}
+  const isWithinRestrictedHours = currentDate.hour() <= 8 || currentDate.hour() >= 16
+  const isWithinFamiliarisationPeriod = currentDate.isAfter(checkWindow.familiarisationCheckStartDate) && currentDate.isBefore(checkWindow.familiarisationCheckEndDate)
+  const isWithinLivePeriod = currentDate.isAfter(checkWindow.checkStartDate) && currentDate.isBefore(checkWindow.checkEndDate)
+  // Familiarisation data
+  pinGenerationEligibilityData.familiarisationCheckStartDate = checkWindow.familiarisationCheckStartDate
+  pinGenerationEligibilityData.familiarisationCheckEndDate = checkWindow.familiarisationCheckEndDate
+  pinGenerationEligibilityData.isFamiliarisationPinGenerationAllowed = (isWithinFamiliarisationPeriod && !isWithinRestrictedHours) || overridePinGenerationEligibility
```

Comment:
@jon-shipley there are cases below where `isWithinRestrictedHours` is used because the negative case is needed.

Reply:
@GeorgeChatzigiannis - that's not idiomatic English though.  We would usually say 'inside opening hours' or 'outside opening hours' for the hours for a shop for example.  Saying 'within' is similar to 'inside', the 'within' bit sets up an expectation in the English speaker that the hours are the usual hours.  There is not much use of 'restricted hours' expect perhaps short trading days like Sunday.

Various style guides usually suggest naming variables for the positive case only.

@GuyHarwood @danwilliams - can we take a vote on this in case it's just me?

Reply:
After discussion, I would vote for `isWithinOpeningHours`, or similar. This may be slightly better than `isWithinRestrictedHours`, and more obvious in cases like this than `!isWithinRestrictedHours`, which I think is the core point here. However, I don't have strong feelings on this, and if George feels the word/concept "restricted" needs additional specific focus, then I'm also not against leaving it as it is.